### PR TITLE
RELATED: TMA-353 User brick - create user groups when they don't exist

### PR DESCRIPTION
**Pull request by vhtien**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/420/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/358/ : FAILURE in 42m 02s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/90/ : FAILURE in 12m 49s

File: lib/gooddata/models/project.rb [hunk 1]
```diff
@@ -1617,10 +1617,16 @@ def remove_users(list)
       end
     end
 
-    def check_groups(specified_groups)
+    def check_groups(specified_groups, options = {})
       groups = user_groups.map(&:name)
       missing_groups = specified_groups - groups
-      fail "All groups have to be specified before you try to import users. Groups that are currently in project are #{groups.join(',')} and you asked for #{missing_groups.join(',')}" unless missing_groups.empty?
+      if options[:create_non_existing_user_groups]
```
Comment (panjan) [inline lib/gooddata/models/project.rb]: Can we have a test for this case so that the bug never occurs again?

Comment (vhtien) [inline lib/gooddata/models/project.rb]: added test for it

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/428/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/366/ : FAILURE in 22m 35s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/98/ : FAILURE in 2m 00s

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/429/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/367/ : FAILURE in 19s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/99/ : FAILURE in 1m 30s

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/432/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/370/ : FAILURE in 21s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/102/ : FAILURE in 2m 27s

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/436/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/374/ : SUCCESS in 42m 20s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/106/ : FAILURE in 12m 58s

**Pull request merged**